src/psyclone/parse/file_info.py

```diff
@@ -78,18 +81,14 @@ class FileInfo:
     - it stores the original filename
     - it will read the source of the file and cache it
     - it will parse it with fparser and cache it
-    - it will parse it with psyir (depends on TODO #2786 "and cache it")
-
-    """
-    def __init__(self, filepath, use_caching: bool = False):
-        """Constructor
+    - it will construct the PSyIR (depends on TODO #2786 "and cache it")
 
-        :param filepath: Path to the file that this
-                         object holds information on.
-        :param use_caching: Use caching of intermediate representations
-        :type filepath: str
+    :param filepath: Path to the file that this
+        object holds information on.
+    :param use_caching: Use caching of intermediate representations
 
-        """
+    """
+    def __init__(self, filepath: str, use_caching: bool = False):
 
         # Full path to file
         self._filename: str = filepath
```

Comment:
It's so far just the File meta information, not the file itself. It's only accessed if needed.
The file is accessed only once the source / psyir / fparser is requested.

You're right that this also supports adding a file which actually doesn't exists.
E.g., to attach a PSyIR tree not read from a file, but which should be attached (as part of a workaround) to the module manager.

Hence, I don't see it necessary to check for the file existance here. If it doesn't exist, this will be triggered once the file is read.